rvgo/fast/memory_test.go

```diff
@@ -173,3 +399,126 @@ func TestMemoryJSON(t *testing.T) {
 	m.GetUnaligned(8, dest[:])
 	require.Equal(t, uint8(123), dest[0])
 }
+
+const (
+	smallDataset  = 1_000
+	mediumDataset = 100_000
+	largeDataset  = 1_000_000
+)
+
+func BenchmarkMemoryOperations(b *testing.B) {
```

Comment:
I moved the benchmarking to a new file: `memory_benchmarking_test.go`. 
I'm not sure if I understand why you mean by `implemented as test`. If you're talking about `testing.B`, it's because the benchmarking capabilities are offered under testing package. Also, there is no validation in the benchmarking test, we only measure the function calls themselves.